addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py

```diff
@@ -386,7 +386,7 @@ def _invoice_constraints_cen_en16931_ubl(self, invoice, vals):
                 constraints.update({'cen_en16931_tax_line': _("Each invoice line shall have one and only one tax.")})
 
         for role in ('supplier', 'customer'):
-            constraints[f'cen_en16931_{role}_country'] = self._check_required_fields(vals[role], 'country_id')
+            constraints[f'cen_en16931_{role}_country'] = self._check_required_fields(vals[role].commercial_partner_id, 'country_id')
```

Comment:
it would probably be better to check the values inside of the xml (using the content of `vals`) rather (path: `cac:Party/cac:PostalAddress/cac:Country/cbc:IdentificationCode`) than the odoo object.